### PR TITLE
PAYARA-1797 Virtual Server Config Cluster Propagation

### PR DESCRIPTION
**Pull request by MattGill98**
- Added update-application-ref command.
- Added update-resource-ref command.
- Updated admin console to use new commands.

### CODE REVIEW

**Comment by MattGill98** [comment]
jenkins quick test please

File: nucleus/deployment/admin/src/main/java/org/glassfish/deployment/admin/UpdateApplicationRefCommand.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    @Override
+    public void execute(AdminCommandContext context) {
+        final ActionReport report = context.getActionReport();
+        final Logger logger = context.getLogger();
+
+        // Make a list of all ApplicationRefs that need to change
+        List<ApplicationRef> applicationRefsToChange = new ArrayList<>();
+
+        // Add the ApplicationRef which is being immediately targetted
+        {
```
Comment (arjantijms) [inline nucleus/deployment/admin/src/main/java/org/glassfish/deployment/admin/UpdateApplicationRefCommand.java]: Could become a separate method with a name based on "Add the ApplicationRef which is being immediately targetted"

File: nucleus/deployment/admin/src/main/java/org/glassfish/deployment/admin/UpdateApplicationRefCommand.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+            ApplicationRef primaryApplicationRef = domain.getApplicationRefInTarget(name, target);
+            if (primaryApplicationRef == null) {
+                report.failure(logger, LOCAL_STRINGS.getLocalString("appref.not.exists", "Target {1} does not have a reference to application {0}.", name, target));
+                return;
+            }
+            applicationRefsToChange.add(primaryApplicationRef);
+        }
+
+        // Add the implicitly targetted ApplicationRefs if the target is in a cluster
+        {
```
Comment (arjantijms) [inline nucleus/deployment/admin/src/main/java/org/glassfish/deployment/admin/UpdateApplicationRefCommand.java]: Could become a separate method as well with name based on "Add the implicitly targetted ApplicationRefs if the target is in a cluster"

**Comment by Pandrex247** [comment]
jenkins test please

**Pull request merged**